.github/workflows/build.yaml

```diff
@@ -0,0 +1,28 @@
+name: Build
+
+on:
+  workflow_call:
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup LXD
+        uses: whywaita/setup-lxd@v1
+        with:
+          lxd_version: latest/stable
+
+      - name: Install rockcraft
+        run: |
+          sudo snap install --classic --channel edge rockcraft
+      - name: Build ROCK
+        run: rockcraft pack --verbose
```

Comment:
Please use the `canonical/craft-actions/rockcraft-pack@main` workflow. You can see an example [here](https://github.com/canonical/magma-orc8r-controller-rock/blob/main/.github/workflows/build.yaml)

Reply:
overall you can very much re-use the workflows from the project linked above (also applicable to `push`)